### ISSUE TITLE
mypy doesn't detect the following error: `TypeError: unsupported operand type(s) for |: 'str' and 'str'`

### ISSUE THREAD

**Issue by soufianeamini** [python/mypy#18827]
While using mypy to check for code generated by an LLM, it generated code that boiled down to the following:
```python
from typing import Literal

A = Literal["A" |  "B"] # doesn't detect the error
# B = Literal["A" *  "B"] # mypy rightfully detects these ones
# C = Literal["A" -  "B"] # mypy rightfully detects these ones
```

**Bug Report**

mypy doesn't detect an error that it detects for other operators.

**To Reproduce**

Run mypy through this code and see it not detect the error.
```python
from typing import Literal

A = Literal["A" |  "B"] # doesn't detect the error
# B = Literal["A" *  "B"] # mypy rightfully detects these ones
# C = Literal["A" -  "B"] # mypy rightfully detects these ones
```

**Expected Behavior**

Get an error similar to the ones for `*` and `-`, like `Unsupported operand types for | ("str" and "str") [operator]`

**Actual Behavior**

Success: no issues found in 1 source file

- Mypy version used: 1.15
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: Python 3.13.2

**Comment by brianschubert** [comment]
Thanks for the report! Closing as duplicate of #17260
